pandas/sparse/array.py

```diff
@@ -769,14 +769,17 @@ def make_sparse(arr, kind='block', fill_value=None):
     if isnull(fill_value):
         mask = notnull(arr)
     else:
-        mask = arr != fill_value
+        mask = np.not_equal(arr, fill_value)
+        # In NumPy 1.12.0, not implemented at least for arr.dtype=str
+        if mask is NotImplemented:
+            mask = np.not_equal(arr.astype(object), fill_value)
```

Comment:
Using `np.not_equal()` as an alternative to `!=`. Operator `!=` returns `False` comparing `arr` when `dtype=str` and `0`, issuing a FutureWarning that eventually, comparison will be element-wise. `np.not_equal()` doesn't issue such a warning but returns `NotImplemented`. One works with what one has. :smiley: 

So if I just want element-wise comparison every time, regardless of dtype, I'd better do:
```py
if pd.types.common.is_string_dtype(arr):
   arr = arr.astype(object)

mask = arr != fill_value
...
```
?